packages/block-editor/src/components/inserter/index.native.js

```diff
@@ -301,6 +315,7 @@ export default compose( [
 			getBlockOrder,
 			getBlockIndex,
 			getBlock,
+			getSettings,
```

Comment:
Hmm, just to be sure, is this change (pulling up the `getSettings` destructuring) related to the tooltip feature, or just an opportunity to clean up the code in `getDefaultInsertionIndex()`? I see the change introduced in https://github.com/WordPress/gutenberg/commit/56dcae9110af78ae6e6ae19b55cf95f219e2bd16 but I feels not really related? 🤔

Reply:
`getSettings` was hoisted higher up as it is now used in two places ([preexisting](https://github.com/WordPress/gutenberg/blob/1808871ceedcd07198eddf682686929ab5ed03b0/packages/block-editor/src/components/inserter/index.native.js#L339-L341), [new](https://github.com/WordPress/gutenberg/blob/1808871ceedcd07198eddf682686929ab5ed03b0/packages/block-editor/src/components/inserter/index.native.js#L381-L382)) due the need to access `canViewEditorOnboarding` in this file. My (untested) inclining was that "selecting" `getSettings` multiple times was possibly poorer performance. 

However, as you pointed out in https://github.com/WordPress/gutenberg/pull/32001#commitcomment-51125756, I overlooked actually using the new, destructured reference. 😄 I will update this file so both use the destructured reference, unless you'd prefer me to revert the destructuring. I do not have strong feelings on this topic. I'm sure the performance gain, if any, is marginal.

Reply:
Ah, I see. 👍 for going with the hoisted higher up destructured reference.

> I will update this file

While at it, wdyt about the other point in [my comment](https://github.com/WordPress/gutenberg/pull/32001#commitcomment-51125756) about renaming at the spot to something slightly more descriptive?

Reply:
@hypest sure, happy to rename if it is helpful. However, there is [a preexisting `getEditorSettings`](https://github.com/WordPress/gutenberg/blob/6a09db457c0f3f937e214ccfd2a970dab81f24d0/packages/edit-post/src/components/header/header-toolbar/index.native.js#L119) that is used, albeit only once, alongside `getSettings` as well. Do you think the name `getEditorSettings` still makes sense in this file?

Reply:
> there is a preexisting getEditorSettings that is used

Good point. Looks like it's pulled from a different store than in our case here, maybe we can use the store name to make it more unique/specific: wdyt about naming it `getBlockEditorSettings()` here?

Reply:
`getBlockEditorSettings` works for me! Thank you.